plugins/modules/rds_instance.py

```diff
@@ -1239,17 +1351,30 @@ def main():
     method_name = get_rds_method_attribute_name(instance, state, module.params['creation_source'], module.params['read_replica'])
 
     if method_name:
+
+        # Exit on create/delete if check_mode
+        if module.check_mode and method_name in ['create_db_instance', 'delete_db_instance']:
+            module.exit_json(changed=True, **camel_dict_to_snake_dict(instance, ignore_list=['Tags', 'ProcessorFeatures']))
+
         raw_parameters = arg_spec_to_rds_params(dict((k, module.params[k]) for k in module.params if k in parameter_options))
-        parameters = get_parameters(client, module, raw_parameters, method_name)
+        parameters_to_modify = get_parameters(client, module, raw_parameters, method_name)
 
```

Comment:
I would probably also add a warning when the user sets `allow_major_version_upgrade` parameter with a descriptive message ('The idempotency is not guaranteed whith allow_major_version_upgrade' or something that makes more sense).

Reply:
So I was going to do this, but it only fails on check mode (since the rds util will catch and handle no changes _actually_ being made and return changed=False). I guess I can add the warning when it's specified during a check mode run

Reply:
Not sure I get the problem. I guess something like this should work:

if modules.params.get(allow_major_version_upgrade):
    module.warns('Exhaustive Message here')

Reply:
the problem only exists when run in check mode - ill update with a warning in that scenario. thanks!